Lib/test/test_bytes.py

```diff
@@ -1359,6 +1359,33 @@ def by(s):
         b = by("Hello, world")
         self.assertEqual(re.findall(br"\w+", b), [by("Hello"), by("world")])
 
+    def test_resize(self):
+        ba = bytearray(b'abcdef')
+        self.assertIsNone(ba.resize(3))
+        self.assertEqual(ba, bytearray(b'abc'))
+        self.assertIsNone(ba.resize(10))
+        self.assertEqual(len(ba), 10)
+        self.assertEqual(ba[:3], bytearray(b'abc'))
```

Comment:
Can you test bytes after abc? Check that resize() fills the buffer with null bytes.

Reply:
👍 , ba[:4] will be: b'abc\0', everything after 4th byte is left to surrounding allocator to set until get to the last byte which will be `\0`

Reply:
For my particular system it is: `bytearray(b'abc\x00ef\x00\xcd\xcd\xcd')`, I think the `b"ef\x00"` is because the realloc succeeds in place, which doesn't seem good to test for

Reply:
Memory is not initialized? That sounds scary.

Reply:
Agreed, it's the existing API... Have been watching things like https://discourse.llvm.org/t/making-ftrivial-auto-var-init-zero-a-first-class-option/55143 / always initializing but that seems to be moving out to compilers/systems slowly...

Reply:
Could make the Python wrapper of the C API always do the memset. Would definitely still save memory allocation + copies over other ways of resizing the `bytearray`